File: marathon.c
#include <stdio.h>
#include <stdlib.h>

#include "user.h"

#define DEFAULT_FILE "user.txt"

int main(int argc, char* argv[])
{
  	int number = 0, ret = 0;
	char *datafile = NULL;
	user *p = NULL;

	// Read a filename
	if(argc == 1) {
		datafile = DEFAULT_FILE;
	} else if (argc == 2) {
		datafile = argv[1];
	} else {
		printf("Usage : %s datafile\n", argv[0]);
		exit(-1);
	}

	// Load data from the file
	ret = user_create(datafile, &p, &number);
  	if( ret != 0 ) {
		printf("user_create() returns error! %d\n", ret);
		exit(ret);
	}

	// Display some statistic data
	ret = user_stat(p, number);
	if( ret != 0 ) {
		printf("user_stat returns() error (errno: %d)\n", ret);
	}

	// Close the file data
	user_close();

	return 0;
}

File: marathon.java
//190606

package project;

import java.io.File;
import java.util.Scanner;

//import project.user.*;

public class marathon {

	static String DEFAULT_FILE  = "userinfo.txt";

	public static void main(String[] args) {
		{
			  String datafile;

			  if(args.length == 1)
			  {
			    datafile = DEFAULT_FILE;
			  } else if(args.length == 2)
			  {
			    datafile = args[1];
			  } else if(args.length >2)
			  {
			    System.out.println("Usage: ./marathon datafile");
			    System.exit(0);
			  }


					try
					{
					String path = marathon.class.getResource("").getPath();
					File file = new File(path + "userinfo.txt");
					Scanner scanfile = new Scanner(file);
					int number = scanfile.nextInt();
					user[] p = new user[number];
					for(int i = 0;i<number;i++)
					{
						String name = scanfile.next();
						double record = scanfile.nextDouble();
						int height = scanfile.nextInt();
						int weight = scanfile.nextInt();
						int age = scanfile.nextInt();
						char gender = scanfile.next().charAt(0);
						p[i] = new user(name,record,height,weight,age,gender);
						p[i].display();
					}
					user.user_state(p, number);
					scanfile.close();
					}
					catch (Exception ex)
					{
						ex.printStackTrace();
					}
				}



			  return;
			}
	}
